ada-project-docs/wave_04.md

```diff
@@ -0,0 +1,57 @@
+# OPTIONAL Wave 4: Enhancements
+
+These really are **optional** - if you've gotten here and you have time left, that means you're moving speedy fast!
+
+### More Inventory Management
+All these endpoints should support all 3 query parameters. All fields are sortable.
+
+#### `GET /rentals/overdue`
+List all customers with overdue videos
+
+Fields to return:
+- `video_id`
+- `title`
+- `customer_id`
+- `name`
+- `postal_code`
+- `checkout_date`
+- `due_date`
+
+### `GET /customers/<id>/history`
+List the videos a customer has checked out _in the past_. Current rentals should not be included.                                                                                  
+
+URI parameters:
+- `id`: Customer ID
+
+Fields to return:
+- `title`
+- `checkout_date`
+- `due_date`
+
+## Enhancements
+
+### More Query Parameters
+The following 3 _optional_ query parameters:
+
+| Name          | Value   | Description
+|---------------|---------|------------
+| `sort`        | string  | Sort objects by this field, in ascending order
+| `count`       | integer | Number of responses to return per page
+| `page_num`    | integer | Page of responses to return
+
+should additionally be accepted by the following three endpoints:
+- `GET /video`
+- `GET /customers/<id>/history`
+- `GET /videos/<id>/history`
```

Comment:
Sorry if I missed something, but I believe this is the only place I've seen this endpoint referenced. Following the pattern of the rest of the project, I assume this endpoint is to retrieve the customer rental history for the video with `id=<id>`. It might be worth describing the endpoint above with the `/rentals/overdue` and `/customers/<id>/history` endpoints.

Reply:
+1, it looks like the `GET /videos/<id>/history` wasn't moved from Wave 3 to here when we decided to cut it from required waves.